core/block/editor/file/file.go

```diff
@@ -303,6 +306,14 @@ func (sf *sfile) dropFilesSetInfo(info dropFileInfo) (err error) {
 		s.Unlink(info.blockId)
 		return sf.Apply(s)
 	}
+	if layout, ok := sf.Layout(); ok && layout == model.ObjectType_collection {
+		s := sf.NewState()
+		store := s.GetStoreSlice(template.CollectionStoreKey)
+		if !slices.Contains(store, info.file.TargetObjectId) {
```

Comment:
It looks like the store already contains methods to verify if some object is already present. 
https://en.wikipedia.org/wiki/Separation_of_concerns